flatten_data_to_anndata.py

```diff
@@ -0,0 +1,145 @@
+    """
+    try:
+        with open(file_path, "r") as file:
+            data = json.load(file)
+            return data
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        print(f"Error reading JSON file: {e}")
+        return None
+
+
+def read_anndata_file(file_path: str) -> Optional[anndata.AnnData]:
+    """Load anndata object from a file.
+
+    Args:
+        file_path: The path to the file containing the anndata object.
+
+    Returns:
+        The loaded anndata object if successful, else None.
+    """
+    try:
+        anndata_obj = anndata.read_h5ad(file_path, backed="r")
+        return anndata_obj
+    except Exception as e:
+        print(f"An error occurred while loading the file: {e}")
+        return None
+
+
+def is_list_of_strings(var):
+    """
+    Check if a value is a list of strings.
+
+    Parameters:
+        var (list or any): The value to be checked.
+
+    Returns:
+        bool: True if the value is a list containing only string elements,
+              False otherwise.
+
+    """
+    return isinstance(var, list) and all(isinstance(item, str) for item in var)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--json", required=True, help="Input JSON file path")
+    parser.add_argument("--anndata", required=True, help="Input AnnData file path")
+    parser.add_argument(
+        "--output",
+        help="Output AnnData file name (default: output.h5ad)",
+        default="output.h5ad",
+    )
+
+    args = parser.parse_args()
+    json_file_path = args.json
+    anndata_file_path = args.anndata
+    output_file_path = args.output
+
+    input_json = read_json_file(json_file_path)
+    input_anndata = read_anndata_file(anndata_file_path)
+
+    # obs
+    annotations = input_json[ANNOTATIONS]
+
+    for ann in annotations:
+        cell_ids = ann.get(CELL_IDS, [])
+
+        for k, v in ann.items():
+            if k == CELL_IDS:
+                continue
+            key = f"{k}-{ann[LABELSET]}"
+            value = v if not isinstance(v, list) else ", ".join(sorted(v))
+
+            input_anndata.obs[key] = ""
+
+            for index_to_insert in ann[CELL_IDS]:
+                input_anndata.obs.at[index_to_insert, key] = value
+
+    # uns
+    uns_json = {}
+    root_keys = list(input_json.keys())
+    root_keys.remove(ANNOTATIONS)
+    for key in root_keys:
+        value = input_json[key]
+        if is_list_of_strings(value):
+            uns_json[key] = ", ".join(sorted(value))
+        elif isinstance(value, str):
+            uns_json[key] = value
+        else:
+            for labelset in value:
+                for k, v in labelset.items():
+                    if k == LABELSET_NAME:
+                        continue
+                    new_key = f"{labelset.get(LABELSET_NAME, '')}-{k}"
+                    uns_json.update({new_key: v})
+
+    input_anndata.uns.update(uns_json)
+    input_anndata.write(output_file_path)
```

Comment:
If the input and output file path will be the same the error will occur here because the file is open in backed mode. So, please validate that input and output path are different (if it suppose to be) or handle this case.

Reply:
@hkir-dev  - could you make sure this is addressed in https://github.com/cellannotation/cas-tools/pull/8